drivers/power/lt7170/lt7170.c

```diff
@@ -0,0 +1,1074 @@
+			return ret;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Set output voltage and its upper limit
+ * @param dev - Device structure
+ * @param vout_command - Output voltage in millivolts
+ * @param vout_max - Output voltage upper limit in millivolts
+ * @return 0 in case of success, negative error code otherwise
+ */
+int lt7170_vout_value(struct lt7170_dev *dev, int vout_command, int vout_max)
+{
+	int ret;
+
+	ret = lt7170_write_word_data(dev, LT7170_VOUT_COMMAND, vout_command);
+	if (ret)
+		return ret;
+
+	return lt7170_write_word_data(dev, LT7170_VOUT_MAX, vout_max);
+}
+
+/**
+ * @brief Set output voltage transition rate
+ * @param dev - Device structure
+ * @param tr - Transition rate in V/s or mV/ms
+ * @return 0 in case of success, negative error code otherwise
+ */
+int lt7170_vout_tr(struct lt7170_dev *dev, int tr)
+{
+	return lt7170_write_word_data(dev, LT7170_VOUT_TRANSITION_RATE, tr);
+}
+
+/**
+ * @brief Set output voltage margin
+ * @param dev - Device structure
+ * @param margin_high - Upper margin in millivolts
+ * @param margin_low - Lower margin in millivolts
+ * @return 0 in case of success, negative error code otherwise
+ */
+int lt7170_vout_margin(struct lt7170_dev *dev, int margin_high, int margin_low)
+{
+	int ret;
+
+	if (margin_high < margin_low)
+		return -EINVAL;
+
+	ret = lt7170_write_word_data(dev, LT7170_VOUT_MARGIN_HIGH, margin_high);
+	if (ret)
+		return ret;
+
+	return lt7170_write_word_data(dev, LT7170_VOUT_MARGIN_LOW, margin_low);
+}
+
+/**
+ * @brief Set input voltage window at which power conversion will proceed
+ * @param dev - Device structure
+ * @param vin_on - Input voltage in millivolts at which conversion will start
+ * @param vin_off - Input voltage in millivolts at which conversion will stop
+ * @return 0 in case of success, negative error code otherwise
+ */
+int lt7170_set_vin(struct lt7170_dev *dev, int vin_on, int vin_off)
+{
+	int ret;
+
+	if (vin_on < vin_off)
+		return -EINVAL;
+
+	if (vin_on < LT7170_VIN_ON_MIN || vin_on > LT7170_VIN_ON_MAX)
+		return -EINVAL;
+
+	if (vin_off < LT7170_VIN_OFF_MIN || vin_off > LT7170_VIN_OFF_MAX)
+		return -EINVAL;
+
+	ret = lt7170_write_word_data(dev, LT7170_VIN_ON, vin_on);
+	if (ret)
+		return ret;
+
+	return lt7170_write_word_data(dev, LT7170_VIN_OFF, vin_off);
+}
+
+/**
+ * @brief Set timing values
+ * @param dev - Device structure
+ * @param timing_type - Timing value type.
+ * 			Example: LT7170_TON_DELAY_TYPE
+ * 				 LT7170_TON_RISE_TYPE
+ * 				 LT7170_TOFF_DELAY_TYPE
+ * 				 LT7170_RETRY_DELAY_TYPE
+ * @param time - Time value in microseconds
+ * @return 0 in case of success, negative error code otherwise
+ */
+int lt7170_set_timing(struct lt7170_dev *dev,
+		      enum lt7170_timing_type timing_type, int time) // in us
+{
+	return lt7170_write_word_data(dev, timing_type, time);
+}
+
+/**
+ * @brief Set switching frequency
+ * @param dev - Device structure
+ * @param freq - Frequency to set in Hz. Value should be between 400 kHz and
+ * 		 4000 kHz.
```

Comment:
Confusing parameter. Is it Hz or kHz?

Reply:
It is set in Hz. Would it be better if I avoid using 'k' and display the zeros here instead?

Reply:
It is set in Hz. Would it be better if I avoid using 'k' and display the zeros here instead?